storagemarket/deal_execution.go

```diff
@@ -83,6 +83,8 @@ func (p *Provider) execDeal(deal *smtypes.ProviderDealState, dh *dealHandler) (d
 	// Capture any panic as a manually retryable error
 	defer func() {
 		if err := recover(); err != nil {
+			// TODO: it'd be nice to expose and log these errors in tests
+			fmt.Println("panic: ", err, string(debug.Stack()))
```

Comment:
Let's remove this line

Reply:
Why? At the moment if a panic happens in tests it is not visible, because we catch it and recover from it, which is super confusing.

Reply:
I am removing the TODO as the println is pretty much doing what it says.

Reply:
A panic happening here is really a problem with Boost that we should fix, and generally should never happen. I agree it is probably a good idea to catch and recover from panics if we somehow introduce a bug that a single deal manages to trigger, so that we don't bring `boostd` down.